independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/AnalyticsService.java

```diff
@@ -70,17 +70,22 @@ public class AnalyticsService implements AutoCloseable {
 
     final private RestClient restClient;
     final private ConfigService config;
-    final private AnonymousUserId userId;
+    final private AnonymousUserId anonymousUserId;
     final private MessageWriter log;
     final FileLocations fileLocations;
 
     public AnalyticsService(final FileLocations fileLocations, MessageWriter log) {
         this.fileLocations = fileLocations;
-        this.log = log;
+        if (log == null) {
```

Comment:
@brunobat in which case log could be null? Is it not a required argument?

Reply:
It is, I'm just making sure we don't have a null Pointer because of this.

Reply:
Do you know how it could end up being null?

Reply:
No, it shouldn't be. I'm just being extra cautious. I don't want ppl to have erros because of some exotic setup.